File: fixtures/scale/lua/large/checks/check_djbhash.lua
local djbhash = require("ciphers.djbhash")
local djbhash_spec = require("specs.djbhash_spec")
local corpus_hash = require("corpus.corpus_hash")

local function check_djbhash()
  local spec = djbhash_spec.get()
  for _, text in ipairs(corpus_hash.corpus_hash()) do
    local first = djbhash.djbhash_digest(text)
    local second = djbhash.djbhash_digest(text)
    if first ~= second or #first ~= 8 then return false end
  end
  return spec.category == "hash"
end

return { check_djbhash = check_djbhash }

File: fixtures/scale/javascript/large/src/checks/checkDjbhash.js
import djbhash from '../ciphers/djbhash.js';
import djbhashSpecMod from '../specs/djbhashSpec.js';
import corpusHashMod from '../corpus/corpusHash.js';

function checkDjbhash() {
  const spec = djbhashSpecMod.djbhashSpec();
  for (const text of corpusHashMod.corpusHash()) {
    const first = djbhash.djbhashDigest(text);
    const second = djbhash.djbhashDigest(text);
    if (first !== second || first.length !== 8) return false;
  }
  return spec.category === 'hash';
}

export default { checkDjbhash };
